### PR TITLE
ISAPI support

### PR DESCRIPTION
**Pull request by viniciussanchez**

### CODE REVIEW

File: src/Horse.API.pas [hunk 1]
```diff
@@ -0,0 +1,162 @@
+unit Horse.API;
```
Comment (hunsche) [inline src/Horse.API.pas]: Raname file to Horse.Core

File: src/Horse.CGI.pas [hunk 2]
```diff
@@ -0,0 +1,74 @@
+unit Horse.CGI;
```
Comment (hunsche) [inline src/Horse.CGI.pas]: Rename file to Horse for compatibility.

File: src/Horse.CGI.pas [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+interface
+
+uses IdHTTPWebBrokerBridge, Horse.API, IdContext;
+
```
Comment (hunsche) [inline src/Horse.CGI.pas]: Added shortcut in uses of THorseRequest and THorseResponse, same as Horse.Core file.

File: src/Horse.CGI.pas [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  FPort := APort;
+end;
+
+constructor THorse.Create;
+begin
+  inherited Create;
+  FPort := DEFAULT_PORT;
+end;
+
+procedure THorse.OnAuthentication(AContext: TIdContext; const AAuthType, AAuthData: String; var VUsername, VPassword: String;
```
Comment (hunsche) [inline src/Horse.CGI.pas]: Test if it is necessary.

File: src/Horse.Commons.pas [hunk 5]
```diff
@@ -0,0 +1,364 @@
...
+
+type
+  /// <summary>
+  ///   HTTP Status Codes
+  ///   httpstatuses.com is an easy to reference database of HTTP Status Codes
+  ///   with their definitions and helpful code references all in one place.
+  ///   Visit an individual status code via httpstatuses.com/code or browse the
+  ///   list below.
+  /// </summary>
+  HTTP_STATUS = record
```
Comment (hunsche) [inline src/Horse.Commons.pas]: Normalize to PascalCase, sample HTTPStatus.

Comment (hunsche) [inline src/Horse.Commons.pas]: Test change record to type.

File: src/Horse.HTTP.pas [hunk 6]
```diff
@@ -19,19 +16,18 @@   THorseRequest = class
...
-    procedure InitializeHeaders;
+    function GetHeaders(AIndex: string): string;
   public
     function Body: string; overload;
     function Body<T: class>: T; overload;
     function Session<T: class>: T;
     function Query: THorseList;
     function Params: THorseList;
-    function Headers: THorseList;
+    property Headers[index: string]: string read GetHeaders;
```
Comment (hunsche) [inline src/Horse.HTTP.pas]: Change to behavior of THorseList (TDictionary).

File: src/Horse.ISAPI.pas [hunk 7]
```diff
@@ -0,0 +1,28 @@
+unit Horse.ISAPI;
+
+interface
+
+uses Horse.API;
+
```
Comment (hunsche) [inline src/Horse.ISAPI.pas]: Added shortcut in uses of THorseRequest and THorseResponse, same as Horse.Core file.

File: src/Horse.CGI.pas (hunk 4 above)
Comment (viniciussanchez) [reply src/Horse.CGI.pas]: image

**Pull request merged**